### PR TITLE
compressor/zstd: add zstd compression plugin

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
retest this please

**Comment by tchaikov** [comment]
retest this please

**Comment by liewegas** [comment]
retest this please

**Comment by tchaikov** [comment]
retest this please.

**Comment by tchaikov** [comment]
```
/home/jenkins-build/build/workspace/ceph-pull-requests/qa/workunits/cephtool/test.sh:32: expect_false:  ceph osd pool set real-tier cache_target_dirty_ratio -.2
Traceback (most recent call last):
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/build/bin/ceph", line 106, in <module>
    import rados
ImportError: libceph-common.so: cannot map zero-fill pages: Cannot allocate memory
```

retest this please.

**Comment by liewegas** [comment]
this needs a small fix, see 

https://github.com/ceph/ceph-ci/commit/wip-sage-testing

**Comment by tchaikov** [comment]
@liewegas included f8ef7f657407ef65c340b8583e1f5119ce78bd4f .

**Comment by liewegas** [comment]
the debian build fails:
<pre>
-- Installing: /build/ceph-11.1.0-6995-gf8ef7f6/debian/tmp/usr/lib/ceph/compressor/libceph_zlib.so.2.0.0
-- Installing: /build/ceph-11.1.0-6995-gf8ef7f6/debian/tmp/usr/lib/ceph/compressor/libceph_zlib.so.2
-- Installing: /build/ceph-11.1.0-6995-gf8ef7f6/debian/tmp/usr/lib/ceph/compressor/libceph_zlib.so
-- Installing: /build/ceph-11.1.0-6995-gf8ef7f6/debian/tmp/usr/local/include/zstd.h
-- Installing: /build/ceph-11.1.0-6995-gf8ef7f6/debian/tmp/usr/local/include/zbuff.h
-- Installing: /build/ceph-11.1.0-6995-gf8ef7f6/debian/tmp/usr/local/include/zdict.h
-- Installing: /build/ceph-11.1.0-6995-gf8ef7f6/debian/tmp/usr/local/lib/libzstd.a
CMake Error at src/compressor/zstd/lib/cmake_install.cmake:78 (file):
  file INSTALL cannot find
  "/build/ceph-11.1.0-6995-gf8ef7f6/obj-x86_64-linux-gnu/src/compressor/zstd/lib/libzstd.1.1.1024.so".
Call Stack (most recent call first):
  src/compressor/zstd/cmake_install.cmake:37 (include)
  src/compressor/cmake_install.cmake:39 (include)
  src/cmake_install.cmake:387 (include)
  cmake_install.cmake:37 (include)
  

</pre>

File: src/compressor/zstd/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# zstd
+
+# libzstd - build it statically
+list(APPEND ZSTD_CMAKE_ARGS
+  -DCMAKE_C_COMPILER_ARGS="-fPIC -Wno-unused-variable -O3")
```
Comment (cbodley) [inline src/compressor/zstd/CMakeLists.txt]: from the build log:
```
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_C_COMPILER_ARGS
```

Comment (tchaikov) [inline src/compressor/zstd/CMakeLists.txt]: @cbodley fixed and repushed.

**Comment by tchaikov** [comment]
changelog

- do not set the build directory to "current binary dir" of "src/compressor/zstd". by sharing the build directory, "make install" will always tries to use the libzstd's Makefile created by its CMakeLists.txt. this prevents the `INSTALL_COMMAND "true"` overriding for working.

**Comment by tchaikov** [comment]
```
/home/jenkins-build/build/workspace/ceph-pull-requests/qa/workunits/ceph-helpers.sh:1151: wait_for_clean:  return 1
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/erasure-code/test-erasure-eio.sh:137: rados_put_get:  return 1
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/erasure-code/test-erasure-eio.sh:172: rados_get_data_eio:  return 1
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/erasure-code/test-erasure-eio.sh:313: TEST_rados_get_with_subreadall_eio_shard_0:  return 1
/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/erasure-code/test-erasure-eio.sh:38: run:  return 1
```
https://jenkins.ceph.com/job/ceph-pull-requests/17475/consoleFull#27421202540526d21-3511-427d-909c-dd086c0d1034
retest this please.

**Comment by tchaikov** [comment]
```
Assert because (miter != peer_missing.end())
Indication 2 
```
test failed due to a known issue already got fixed on master.

**Comment by tchaikov** [comment]
rebased against master.

**Pull request merged**